File: tree/dfs.java

package Tree;

class Node {
  int key;
  Node left;
  Node right;

  Node(int k) {
    key = k;
    left = right = null;
  }
}

class dfs {

  public static void main(String args[]) {
    Node root = new Node(10);
    root.left = new Node(20);
    root.right = new Node(30);
    root.right.left = new Node(40);
    root.right.right = new Node(50);

    inorder(root);
    preorder(root);
    postorder(root);
  }

  public static void inorder(Node root) {
    if (root != null) {
      inorder(root.left);
      System.out.print(root.key + " ");
      inorder(root.right);
    }
  }

  public static void preorder(Node root) {
    if (root != null) {
      System.out.print(root.key + " ");
      preorder(root.left);
      preorder(root.right);
    }
  }

  public static void postorder(Node root) {
    if (root != null) {
      postorder(root.left);
      postorder(root.right);
      System.out.print(root.key + " ");
    }
  }
}

File: tree/dfs.cpp
#include <bits/stdc++.h>
using namespace std;

struct Node
{
  int key;
  struct Node *left;
  struct Node *right;
  Node(int k)
  {
    key = k;
    left = right = NULL;
  }
};

void inorder(Node *root)
{
  if (root == NULL)
    return;
  inorder(root->left);
  cout << root->key << " ";
  inorder(root->right);
}

void preorder(Node *root)
{
  if (root == NULL)
    return;
  cout << root->key << " ";
  preorder(root->left);
  preorder(root->right);
}

void postorder(Node *root)
{
  if (root == NULL)
    return;
  postorder(root->left);
  postorder(root->right);
  cout << root->key << " ";
}
int main()
{

  Node *root = new Node(10);
  root->left = new Node(20);
  root->right = new Node(30);
  root->left->left = new Node(40);
  inorder(root);
  preorder(root);
  postorder(root);
}